rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,501 @@
+
+**Considering**: General reference logic between doc-comments and expressions.
+
+**Decision**: Doc-comments refer to the **immediately** preceded expression. Only whitespaces are allowed in between.
+
+> Note: We are working on a generic example implementation that covers about 90% of all documentation use cases.
+
+<details>
+<summary>Arguments</summary>
+
+- (+) Doc-comments should have only one variant to reduce complexity. Referencing the next node seems straightforward. 
+- (-) A variant that references the previous node in the AST should be avoided to reduce complexity.
+- (+) Relation between documentation and the referenced implementation is straightforward and back-trackable.
+- (-) Concrete Implementation might be complex.
+- (-) Unclear if complete documentation might also need backward references. 
+    - (-), e.g., rust uses backward references only at the top of file comments.
+    - (+) Nix files can only have ONE expression; the next AST Node, in case of top-of-file comments, is thus always only that one expression. (Unlike in Rust)
+- (-) Tools need to be smart enough to understand asignments `=` and other forms of creating names for anonymous expressions. (e.g., `callPackage` and `import` )
+    - (+) Tools can still come up with other solutions that do not involve calculating everything dynamically from nix code but could also involve a static configuration.
+    - (+) The whole `tool point` is an implementation detail as long as it is not impossible. The current tool, `nixdoc`, already proves that it is possible to have static documentation to a certain degree.
+
+</details>
+
+## Single-line doc-comments (do not exist)
+
+**Observing**: Nix offers two variants of comments; single- (`#`) and multi-line comments (`/* */`). There may be use cases where it is desirable to have a form of single-line comments subtyped for doc-comment purposes. 
+
+**Considering**: Single-line comment for documentation.
+
+**Decision**: Single-line comments **cannot be used** in any form for documentation puposes.
+
+<details>
+<summary>Arguments</summary>
+
+- (-) Doc-comments should have only one variant to reduce complexity.  
+- (-) documentation will likely take up more than one line.
+- (-) If documentation grows bigger than one line, refactoring into a multiline-doc-comment must occur.
+- (+) It Would be consistent with providing variants for both nix comments.
+- (+) Offer the choice.
+- (o) Single lines can also be concatenated to form multi-line documentation.
+- (+) Takes up less vertical space
+- (-) Visually confusing when every line starts with a reserved character.
+    - (-) Potential visual conflicts with the content
+- (+) Indentation of the content is clear.
+
+</details>
+
+## Doc-comment examples
+
+The following examples demonstrate the concrete usage scenarios:
```

Comment:
Not sure what is difference between this and  `## Examples`` section.

If it is section for generic example, then we probably should indicate that without saying "concrete usage scenarios" as there already section for "emphasize the previously discussed characteristics" that seems to be demonstrating concrete examples.

Reply:
Not sure either ;) It seems i have duplicated the headings. Thanks